src/mouse_events.cpp

```diff
@@ -785,17 +789,26 @@ void mouse_handler::select_or_action(bool browse)
 	wb::future_map_if_active planned_unit_map;
 
 	if(game_lua_kernel* lk = pc_.gamestate().lua_kernel_.get()) {
-		lk->select_hex_callback(last_hex_);
+		lk->select_hex_callback(clicked_hex);
+		// The callback may modify clicked_hex, so make sure it's still valid
+		if(!pc_.get_map().on_board(clicked_hex)) {
+			return;
+		}
 	}
 
-	unit_map::iterator clicked_u = find_unit(last_hex_);
+	unit_map::iterator clicked_u = find_unit(clicked_hex);
 	unit_map::iterator selected_u = find_unit(selected_hex_);
 
 	if(clicked_u && (!selected_u || selected_u->side() != side_num_ ||
 	  (clicked_u->side() == side_num_ && clicked_u->id() != selected_u->id()))
```

Comment:
What is this `if` meant to check? I think some of the intended logic would check whether both units are on the same side, but the implemented logic is checking if (they're on the same side and also that side is the currently active side).

I'd speculate that should be checking if they're enemies of each other instead, so that attack predictions can be shown.

Reply:
This may be a good question but it's not really in scope for this PR…